jlmkr.py

```diff
@@ -378,6 +380,9 @@ def passthrough_nvidia(
     # while mounting the symlink will be resolved and nvidia-smi will appear as a regular file
     nvidia_files.add("/usr/bin/nvidia-smi")
 
+    # Use current module dir
+    nvidia_files.add("/usr/lib/x86_64-linux-gnu/nvidia/current")
```

Comment:
We're now hardcoding `/usr/lib/x86_64-linux-gnu/nvidia/current`, which is what I was doing in the past for the contents of the `/etc/ld.so.conf.d/jlmkr-nvidia.conf` file as you can see here: https://github.com/Jip-Hop/jailmaker/commit/1d6a7b910e22fb1db8faf0aea8d3a98910f61777

Thanks to the PR from @TrueJournals we no longer had to hardcode `/usr/lib/x86_64-linux-gnu/nvidia/current`, but we could discover it dynamically.

I think this time the solution should be to add the directories in`library_folders` as bind mounts (and then filter out, or not add, the files which are children of any of these `library_folders` since they will already be mounted as part of the library folder). I think currently the `library_folders` only contains `/usr/lib/x86_64-linux-gnu/nvidia/current`. Does that make sense?

Reply:
No, you lost me lol. 
This isn't touching ldconf is it?

Reply:
What I'm trying to say is I rather not hardcode this path as it may change in the future. Dynamically discovering it would be better. This work is already done. If I'm not mistaken the path is already inside the `library_folders` variable. Can you confirm that is the case (by printing the contents of `library_folders`)?

Reply:
Gotcha; i've opted to then leave the existing mounts untouched (else we'd need to re-arrange the whole thing to have the ```library_folders``` detection before the current ```nvidia_mounts``` in order to exclude from ```nvidia_mounts``` where they startwith one of the ```library_folders```)

And just added the library as an additional mount command.
Downside to this somewhat lazy implementation is that it does show a long-ish list of bind mounts on startup. But otherwise, it does work, and causes no harm.

Reply:
btw; the issue is caused by software which doesn't utilise the specific version definitions in ldconf; and instead references the baseline files (eg ```*.so``` or ```*.so.1```) since these are not mentioned in the output of ```nvidia-container-cli``` but are symlinks in the ```current``` folder:
```bash
root@gpu:~# ls -l /usr/lib/x86_64-linux-gnu/nvidia/current/
total 79826
lrwxrwxrwx 1 root root       12 Nov  7  2023 libcuda.so -> libcuda.so.1
lrwxrwxrwx 1 root root       20 Nov  7  2023 libcuda.so.1 -> libcuda.so.545.23.08
-rw-r--r-- 1 root root 29453200 Nov  6  2023 libcuda.so.545.23.08
lrwxrwxrwx 1 root root       15 Nov  7  2023 libnvcuvid.so -> libnvcuvid.so.1
lrwxrwxrwx 1 root root       23 Nov  7  2023 libnvcuvid.so.1 -> libnvcuvid.so.545.23.08
-rw-r--r-- 1 root root 10009920 Nov  6  2023 libnvcuvid.so.545.23.08
lrwxrwxrwx 1 root root       26 Nov  7  2023 libnvidia-cfg.so.1 -> libnvidia-cfg.so.545.23.08
-rw-r--r-- 1 root root   274968 Nov  6  2023 libnvidia-cfg.so.545.23.08
lrwxrwxrwx 1 root root       21 Nov  7  2023 libnvidia-encode.so -> libnvidia-encode.so.1
lrwxrwxrwx 1 root root       29 Nov  7  2023 libnvidia-encode.so.1 -> libnvidia-encode.so.545.23.08
-rw-r--r-- 1 root root   252576 Nov  6  2023 libnvidia-encode.so.545.23.08
lrwxrwxrwx 1 root root       17 Nov  7  2023 libnvidia-ml.so -> libnvidia-ml.so.1
lrwxrwxrwx 1 root root       25 Nov  7  2023 libnvidia-ml.so.1 -> libnvidia-ml.so.545.23.08
-rw-r--r-- 1 root root  1992128 Nov  6  2023 libnvidia-ml.so.545.23.08
lrwxrwxrwx 1 root root       19 Nov  7  2023 libnvidia-nvvm.so -> libnvidia-nvvm.so.4
lrwxrwxrwx 1 root root       27 Nov  7  2023 libnvidia-nvvm.so.4 -> libnvidia-nvvm.so.545.23.08
-rw-r--r-- 1 root root 86781944 Nov  7  2023 libnvidia-nvvm.so.545.23.08
lrwxrwxrwx 1 root root       37 Nov  7  2023 libnvidia-ptxjitcompiler.so.1 -> libnvidia-ptxjitcompiler.so.545.23.08
-rw-r--r-- 1 root root 26589472 Nov  6  2023 libnvidia-ptxjitcompiler.so.545.23.08
```

Reply:
Thanks again for your effort :) and glad to finally understand why this issue is happening.

How hard would it be to go for the full/proper implementation? With jailmaker I strive for the best possible ^^

P.S. Since this is a bugix you should increment the last number of the version according to semver. The new version would be 1.4.1 in this case.

Reply:
Effort is low, but also requires time lol 
The main issue was that the current nvidia_files bit traps in a try block, which I'd also have to reimplement around the library_folders instead. Probably not much effort though, will tweak it whilst amending the version too

Reply:
From the top of my head. Maybe we don't need to filter out these library files if they don't end up in that list in the first place? I know the output of `nvidia-container-cli` can be filtered. We use it to output only the libraries for example. Maybe it can also output everything except for libraries?

Reply:
Note: I looked at moving stuff around, and it doesn't work out as simply as i'd imagined because of how the order of use for some of the vars.

So i've been digging based on what you said about ```nvidia-container-cli``` instead; it doesn't have any explicit options for that style of output directly...
But one option is to use it to list the libraries and then exclude that from it's own full list:
```
nvidia-container-cli list | grep -vFf <(nvidia-container-cli list --libraries)
```
Which I have implemented here in the latest commit. Seems like the most graceful approach, even if the bash call is a bit gross.

Thoughts?

Reply:
@Jip-Hop have updated this to be more pythonic, and move the ```nvidia_libraries``` def to the top to be used both it's existing part (path parent) and also for exclusions.

Reply:
This looks great! Will test later :)